File: src/main/java/com/itsci/mju/maebanjumpen/entity/Housekeeper.java
package com.itsci.mju.maebanjumpen.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.HashSet;
import java.util.Set;

@Entity
@DiscriminatorValue("housekeeper")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString(callSuper = true)
public class Housekeeper extends Member {

    @Column(name="photo_verify_url")
    private String photoVerifyUrl;

    @Enumerated(EnumType.STRING)
    @Column(name="status_verify")
    private VerifyStatus statusVerify = VerifyStatus.PENDING; // 💡 PENDING คือค่าเริ่มต้น

    @Column(name="rating")
    private Double rating = 0.0;

    @Column(name = "daily_rate")
    private String dailyRate = "0.0"; // หรือค่าเริ่มต้นที่เหมาะสมกับ String

    @OneToMany(mappedBy = "housekeeper", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @ToString.Exclude
    private Set<Hire> hires = new HashSet<>();

    @OneToMany(mappedBy = "housekeeper", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @ToString.Exclude
    private Set<HousekeeperSkill> housekeeperSkills = new HashSet<>();

    public enum VerifyStatus {
        PENDING, APPROVED, REJECTED, VERIFIED, NOT_VERIFIED
    }
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/entity/Housekeeper.kt
package com.itsci.mju.maebanjumpen.entity

import jakarta.persistence.*

@Entity
@DiscriminatorValue("housekeeper")
open class Housekeeper(
    @Column(name = "photo_verify_url")
    open var photoVerifyUrl: String? = null,

    @Enumerated(EnumType.STRING)
    @Column(name = "status_verify")
    open var statusVerify: VerifyStatus? = VerifyStatus.PENDING,

    @Column(name = "rating")
    open var rating: Double? = 0.0,

    @Column(name = "daily_rate")
    open var dailyRate: String? = "0.0",

    @OneToMany(mappedBy = "housekeeper", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    open var hires: MutableSet<Hire> = mutableSetOf(),

    @OneToMany(mappedBy = "housekeeper", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    open var housekeeperSkills: MutableSet<HousekeeperSkill> = mutableSetOf()
) : Member() {

    enum class VerifyStatus {
        PENDING, APPROVED, REJECTED, VERIFIED, NOT_VERIFIED
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Housekeeper) return false
        if (!super.equals(other)) return false
        return true
    }

    override fun hashCode(): Int = super.hashCode()
}

